Sources/SentrySwiftUI/SentryTracedView.swift

```diff
@@ -34,41 +34,61 @@ import SentryInternal
 ///         //The part of your content you want to measure
 ///     }.sentryTrace("My Awesome Screen")
 ///
-///
 @available(iOS 13, macOS 10.15, tvOS 13, watchOS 6.0, *)
 public struct SentryTracedView<Content: View>: View {
-    
+    @State var viewWasPresented = false
+
     let content: () -> Content
     let name: String
-    let id: SpanId
-    
+    let nameSource: SentryTransactionNameSource
+
     public init(_ viewName: String? = nil, content: @escaping () -> Content) {
         self.content = content
         self.name = viewName ?? SentryTracedView.extractName(content: Content.self)
-        id = SentryPerformanceTracker.shared.startSpan(withName: self.name,
-                                                       nameSource: viewName == nil ? .component : .custom,
-                                                       operation: "ui.load")
+        self.nameSource = viewName == nil ? .component : .custom
     }
-    
+
     private static func extractName(content: Any) -> String {
         var result = String(describing: content)
-        
+
         if let index = result.firstIndex(of: "<") {
             result = String(result[result.startIndex ..< index])
         }
-        
+
         return result
     }
-    
+
     public var body: some View {
+        if viewWasPresented {
+            return self.content().onAppear()
+        }
+
+        var transactionCreated = false
+        if SentryPerformanceTracker.shared.activeSpanId() == nil {
+            transactionCreated = true
+            let transactionId = SentryPerformanceTracker.shared.startSpan(withName: self.name, nameSource: nameSource, operation: "ui.load")
+            SentryPerformanceTracker.shared.pushActiveSpan(transactionId)
+
+            //According to Apple's documentation, the call to `body` needs to be fast
+            //and can be made many times in one frame. Therefore they don't use async code to process the view.
+            //Scheduling to finish the transaction at the end of the main loop seems the least hack solution right now.
+            DispatchQueue.main.async {
```

Comment:
`m`: I don't understand why we can't use 
```
let result = self.content().onAppear {
    SentryPerformanceTracker.shared.finishSpan(id)
}
```
anymore.

Reply:
Because using `onAppear` to finish the transaction don't reflect the actual performance of the `body` property.
Other factors may impact the time it takes to appear and the user will be mislead to believe this view is the problem.

Reply:
Couldn't other factors not related to this view delay the execution of this?

Reply:
Sorry, I misunderstood the scope of your question. 
This is regarding the umbrella transaction, but the answer still the same, we cant use `onAppear` to finish the transaction, because this event happens before the end of the ui thread loop, it even may happens before the body of other views get called, which means we may not group in the same transaction every SentryTracedView in the same screen.